bus/bus.go

```diff
@@ -560,6 +574,125 @@ func (b *Bus) addRenewedContract(ctx context.Context, renewedFrom types.FileCont
 	return r, nil
 }
 
+func (b *Bus) compatV2Settings(ctx context.Context) error {
+	// escape early if all settings are present
+	if !errors.Is(errors.Join(
+		b.ss.Setting(ctx, stores.SettingGouging, nil),
+		b.ss.Setting(ctx, stores.SettingPinned, nil),
+		b.ss.Setting(ctx, stores.SettingS3, nil),
+		b.ss.Setting(ctx, stores.SettingUpload, nil),
+	), api.ErrSettingNotFound) {
```

Comment:
Considering that we no longer allow for storing arbitrary settings we could get rid of `api.ErrSettingNotFound` and instead have the `SettingsStore` return default settings if not explicitly set. Then we don't need to check for that error just to overwrite it with the defaults and even if users delete settings from the db at runtime directly, `renterd` will correctly fall back to the defaults.

Maybe the settings conversion can then also be a regular database migration.

Reply:
I ended up moving `api.ErrSettingNotFound` to the `sql` package, only the store uses it now. The `SettingStore` now returns default settings if not implicitly set. I considered a regular database migration for the setting migration, I actually even considered that myself when writing it initially, but the migration logic was slightly too complex I found to have in a database migration. Especially because we now fallback to the defaults if the config is not there or invalid (in a F/U PR we also introduce the explorer config which potentially alters the pinned settings).